ios/brave-ios/Sources/BraveWallet/Crypto/Stores/AssetDetailStore.swift

```diff
@@ -456,19 +459,26 @@ class AssetDetailStore: ObservableObject, WalletObserverStore {
       @MainActor group -> [AccountBalance] in
       for accountAssetViewModel in accountAssetViewModels {
         group.addTask { @MainActor in
-          let balance = await self.rpcService.balance(
-            for: token,
-            in: accountAssetViewModel.account,
-            network: network
-          )
-          return [AccountBalance(accountAssetViewModel.account, balance)]
+          var tokenBalance: Double?
+          if accountAssetViewModel.account.coin == .btc {
+            tokenBalance = await self.bitcoinWalletService.fetchBTCBalance(
+              accountId: accountAssetViewModel.account.accountId
+            )
+          } else {
+            tokenBalance = await self.rpcService.balance(
+              for: token,
+              in: accountAssetViewModel.account,
+              network: network
+            )
+          }
+          return [AccountBalance(accountAssetViewModel.account, tokenBalance)]
         }
       }
       return await group.reduce([AccountBalance](), { $0 + $1 })
     }
     for tokenBalance in tokenBalances {
       if let index = accountAssetViewModels.firstIndex(where: {
-        $0.account.address == tokenBalance.account.address
+        $0.account.cacheBalanceKey == tokenBalance.account.cacheBalanceKey
       }) {
         accountAssetViewModels[index].decimalBalance = tokenBalance.balance ?? 0.0
         accountAssetViewModels[index].balance = String(format: "%.4f", tokenBalance.balance ?? 0.0)
```

Comment:
Not specific to Bitcoin, but we need to update `balance` to be a `Double` or `BDouble` on this view model and format as a String in UI code instead of view model. Right now the `totalBalance` computed property uses this string to calculate the total balance, but this value is rounded to 4 decimal places, however total balance is being displayed with 6 decimal places.
So if I have a balance of 0.00001 BTC it will show as 0.000000 BTC total balance.

Since this isn't specific to Bitcoin, I opened a separate issue for this rounding bug:
https://github.com/brave/brave-browser/issues/37670

Reply:
There is actually an idea somehere deep in backlog to return balance in weis, satoshis, lamports, etc. So any rounding happens only when we want to display that balance on screen.
Using double in money related ares is discouraged in general